manage_breast_screening/participants/models.py

```diff
@@ -179,8 +182,23 @@ class Appointment(BaseModel):
     stopped_reasons = models.JSONField(null=True, blank=True)
 
     @cached_property
-    def current_status(self):
-        return self.statuses.first()
+    def current_status(self) -> "AppointmentStatus":
+        """
+        Fetch the most recent status associated with this appointment.
+        If there are no statuses for any reason, assume the default one.
+        """
+        # avoid `first()` here so that `statuses` can be prefetched
+        # when fetching many appointments
+        statuses = list(self.statuses.all())
```

Comment:
Is it worth adding an explicit order here? Would remove some ambiguity even if not strictly necessary.